drivers/clock_control/clock_control_it8xxx2.c

```diff
@@ -0,0 +1,334 @@
+	/*
+	 * PLL frequency setting = 4 (48MHz)
+	 * FND   div = 0 (PLL / 1 = 48 mhz)
+	 * UART  div = 1 (PLL / 2 = 24 mhz)
+	 * SMB   div = 1 (PLL / 2 = 24 mhz)
+	 * SSPI  div = 0 (PLL / 1 = 48 mhz)
+	 * EC    div = 6 (FND / 6 =  8 mhz)
+	 * JTAG  div = 1 (PLL / 2 = 24 mhz)
+	 * PWM   div = 0 (PLL / 1 = 48 mhz)
+	 * USBPD div = 5 (PLL / 6 =  8 mhz)
+	 */
+	/* PLL:48MHz, MCU:48MHz, Fnd:48MHz */
+	[PLL_48_MHZ] = {48000000, 4, 0, 1, 1, 0, 6, 1, 0, 0x05},
+
+	/*
+	 * PLL frequency setting = 7 (96MHz)
+	 * FND   div = 1 (PLL / 2 = 48 mhz)
+	 * UART  div = 3 (PLL / 4 = 24 mhz)
+	 * SMB   div = 3 (PLL / 4 = 24 mhz)
+	 * SSPI  div = 1 (PLL / 2 = 48 mhz)
+	 * EC    div = 6 (FND / 6 = 16 mhz)
+	 * JTAG  div = 3 (PLL / 4 = 24 mhz)
+	 * PWM   div = 1 (PLL / 2 = 48 mhz)
+	 * USBPD div = B (PLL /12 =  8 mhz)
+	 */
+	/* PLL:96MHz, MCU:96MHz, Fnd:48MHz */
+	[PLL_96_MHZ] = {96000000, 7, 1, 3, 3, 1, 6, 3, 1, 0x0B},
+	/* IC Default:  */
+};
+
+void __intc_ram_code chip_pll_ctrl(enum chip_pll_mode mode)
+{
+	struct ecpm_it8xxx2_regs *const ecpm_regs = clock_config->reg_base;
+
+	volatile uint8_t _pll_ctrl __unused;
+
+	ecpm_regs->PLLCTRL = mode;
+	/*
+	 * for deep doze / sleep mode
+	 * This load operation will ensure PLL setting is taken into
+	 * control register before wait for interrupt instruction.
+	 */
+	_pll_ctrl = ecpm_regs->PLLCTRL;
+}
+
+void __intc_ram_code chip_run_pll_sequence(const struct pll_config_t *pll)
+{
+	struct ecpm_it8xxx2_regs *const ecpm_regs = clock_config->reg_base;
+
+	/* Enable HW timer to wakeup chip from the sleep mode */
+	timer_5ms_one_shot();
+	/*
+	 * Configure PLL clock dividers.
+	 * Writing data to these registers doesn't change the
+	 * PLL frequency immediately until the status is changed
+	 * into wakeup from the sleep mode.
+	 * The following code is intended to make the system
+	 * enter sleep mode, and wait HW timer to wakeup chip to
+	 * complete PLL update.
+	 */
+	ecpm_regs->PLLFREQR = pll->pll_setting;
+	/* Pre-set FND clock frequency = PLL / 3 */
+	ecpm_regs->SCDCR0 = (2 << 4);
+	/* JTAG and EC */
+	ecpm_regs->SCDCR3 = (pll->div_jtag << 4) | pll->div_ec;
+	/* Chip sleep after wait for interrupt (wfi) instruction */
+	chip_pll_ctrl(CHIP_PLL_SLEEP);
+	/* Chip sleep and wait timer wake it up */
+	__asm__ volatile  ("wfi");
+	/* New FND clock frequency */
+	ecpm_regs->SCDCR0 = pll->div_fnd << 4;
+	/* Chip doze after wfi instruction */
+	chip_pll_ctrl(CHIP_PLL_DOZE);
+	/* UART */
+	ecpm_regs->SCDCR1 = pll->div_uart;
+	/* SSPI and SMB */
+	ecpm_regs->SCDCR2 = (pll->div_sspi << 4) | pll->div_smb;
+	/* USBPD and PWM */
+	ecpm_regs->SCDCR4 = (pll->div_usbpd << 4) | pll->div_pwm;
+}
+
+static void chip_configure_pll(const struct pll_config_t *pll)
+{
+	struct ecpm_it8xxx2_regs *const ecpm_regs = clock_config->reg_base;
+
+	/* Re-configure PLL clock or not. */
+	if (((ecpm_regs->PLLFREQR & 0x0F) != pll->pll_setting) ||
+	    ((ecpm_regs->SCDCR0 & 0xF0) != (pll->div_fnd << 4)) ||
+	    ((ecpm_regs->SCDCR3 & 0x0F) != pll->div_ec)) {
```

Comment:
Ditto - use define and `BIT_MASK` macro or `GETMASK` macro.

Reply:
@fkokosinski 

Yes, sir. I've updated them as follows:

Definitions:
```
#define EC_FREQ_MASK BIT_MASK(4)
#define FNDFREQ_MASK ~(BIT_MASK(4))
```

Flow control:
```
	/* Re-configure PLL clock or not. */
	if (((ecpm_regs->PLLFREQR & PLLFREQ_MASK) != pll->pll_setting) ||
	    ((ecpm_regs->SCDCR0 & FNDFREQ_MASK) != (pll->div_fnd << 4)) ||
	    ((ecpm_regs->SCDCR3 & EC_FREQ_MASK) != pll->div_ec)) {
```

Thanks again for helps and guidance.